cmd/otel-allocator/internal/watcher/promOperator.go

```diff
@@ -53,7 +53,30 @@ func NewPrometheusCRWatcher(ctx context.Context, logger logr.Logger, cfg allocat
 		return nil, err
 	}
 
-	factory := informers.NewMonitoringInformerFactories(map[string]struct{}{v1.NamespaceAll: {}}, map[string]struct{}{}, mClient, allocatorconfig.DefaultResyncTime, nil) //TODO decide what strategy to use regarding namespaces
+	if len(cfg.PrometheusCR.AllowNamespaces) != 0 && len(cfg.PrometheusCR.DenyNamespaces) != 0 {
+		return nil, fmt.Errorf("both allow and deny namespaces are set, only one should be set")
+	}
```

Comment:
This check should happen during config validation in config.go. Here, we can assume it's valid.

Reply:
I think it's good to do this here too IMO, we do have some users who are asking to be able to self-manage the TA outside of the operator and doing this validation here wouldn't hurt.

Reply:
I agree it should be done in the target allocator application. I just think it should be done when validation configuration rather than when instantiating this struct.

Reply:
ah i misread, apologies :bow: